circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/animation/AnimatedNavDecoration.kt

```diff
@@ -126,16 +126,17 @@ public class AnimatedNavDecoration(
 private fun <T : NavArgument> AnimatedNavDecorator<T, AnimatedNavState>.transitionSpec(
   animatedScreenTransforms: ImmutableMap<KClass<out Screen>, AnimatedScreenTransform>
 ): AnimatedContentTransitionScope<AnimatedNavState>.() -> ContentTransform = spec@{
-  val diff = targetState.backStackDepth - initialState.backStackDepth
-  val sameRoot = targetState.rootScreen == initialState.rootScreen
-  val animatedNavEvent =
-    when {
-      !sameRoot -> AnimatedNavEvent.RootReset
-      diff > 0 -> AnimatedNavEvent.GoTo
-      diff < 0 -> AnimatedNavEvent.Pop
-      // Somehow the back stack has not changed?
-      else -> return@spec EnterTransition.None togetherWith ExitTransition.None
-    }
+
+  val animatedNavEvent = if (targetState.rootScreen != initialState.rootScreen) {
+    AnimatedNavEvent.RootReset
+  } else if (targetState.screen == initialState.screen) {
+    // Target screen has not changed, probably we should not show any animation even if the back stack is changed
+    return@spec EnterTransition.None togetherWith ExitTransition.None
+  } else if (initialState.backStack.contains(targetState.screen)) {
```

Comment:
The `contains` here might not work as expected if we navigate to a screen already in the backstack.
So we probably want to compare the full record using the full backstack (the `args` given to us in `DefaultAnimatedState`), and determine if the initial and target backstacks have diverged.

Reply:
I'm sorry for such late  answer...

I was thinking about how exactly we can/should compare backstacks. And I really think that 'contains' covers really more situations. If target screen was already in stack - we're moving back. In other case screen is new - it's GoTo.

Reply:
Also record itself contains only Screen...

Reply:
Thanks. I will update my PR.